Remove transitive dependency on packr

Checklist:

As discussed here: https://argoproj.slack.com/archives/CASHNF6MS/p1572407437084900?thread_ts=1556732320.124600&cid=CASHNF6MS

**Describe the bug**

The current use of `packr` makes it a transitive dependency for anyone trying to develop against `github.com/argoproj/argo-cd/pkg/apiclient` directly

**To Reproduce**

1. Make a program importing `github.com/argoproj/argo-cd/pkg/apiclient`

2. Run `dep ensure`

3. Compile and run your program

Returns error:
```
panic: stat _vendor_path_/github.com/argoproj/argo-cd/assets/builtin-policy.csv: no such file or directory

goroutine 1 [running]:
github.com/sendgrid/argocd-helper/vendor/github.com/argoproj/argo-cd/util/assets.init.0()
	/Users/ssuter/go/src/github.com/sendgrid/argocd-helper/vendor/github.com/argoproj/argo-cd/util/assets/assets.go:19 +0x2de
```

**Expected behavior**

The program should succeed

**Screenshots**

n/a

**Version**

`1.2.5`

```shell
Paste the output from `argocd version` here.
```

--- a/hack/test.sh
+++ b/hack/test.sh
@@ -1,6 +1,13 @@
 #!/bin/bash
 set -eux -o pipefail
 
+# make sure apiclient does not depend on packr
+which godepgraph || go get github.com/kisielk/godepgraph
+if godepgraph -s github.com/argoproj/argo-cd/pkg/apiclient | grep packr; then
+  echo apiclient package should not depend on packr
+  exit 1
+fi
+
 TEST_RESULTS=${TEST_RESULTS:-test-results}
 
 mkdir -p $TEST_RESULTS

--- a/pkg/apis/application/v1alpha1/types.go
+++ b/pkg/apis/application/v1alpha1/types.go
@@ -30,7 +30,6 @@ import (
 	"github.com/argoproj/argo-cd/util/cert"
 	"github.com/argoproj/argo-cd/util/git"
 	"github.com/argoproj/argo-cd/util/helm"
-	"github.com/argoproj/argo-cd/util/rbac"
 )
 
 // Application is a definition of Application resource.
@@ -1322,9 +1321,6 @@ func (p *AppProject) ValidateProject() error {
 		}
 	}
 
-	if err := p.validatePolicySyntax(); err != nil {
-		return err
-	}
 	return nil
 }
 
@@ -1395,15 +1391,6 @@ func validateGroupName(name string) error {
 	}
 	if invalidChars.MatchString(name) {
 		return status.Errorf(codes.InvalidArgument, "group '%s' contains invalid characters", name)
-	}
-	return nil
-}
-
-// validatePolicySyntax verifies policy syntax is accepted by casbin
-func (p *AppProject) validatePolicySyntax() error {
-	err := rbac.ValidatePolicy(p.ProjectPoliciesString())
-	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "policy syntax error: %s", err.Error())
 	}
 	return nil
 }

--- a/server/project/project.go
+++ b/server/project/project.go
@@ -49,13 +49,25 @@ func NewServer(ns string, kubeclientset kubernetes.Interface, appclientset appcl
 	return &Server{enf: enf, appclientset: appclientset, kubeclientset: kubeclientset, ns: ns, projectLock: projectLock, auditLogger: auditLogger, sessionMgr: sessionMgr}
 }
 
+func validateProject(proj *v1alpha1.AppProject) error {
+	err := proj.ValidateProject()
+	if err != nil {
+		return err
+	}
+	err = rbac.ValidatePolicy(proj.ProjectPoliciesString())
+	if err != nil {
+		return status.Errorf(codes.InvalidArgument, "policy syntax error: %s", err.Error())
+	}
+	return nil
+}
+
 // CreateToken creates a new token to access a project
 func (s *Server) CreateToken(ctx context.Context, q *project.ProjectTokenCreateRequest) (*project.ProjectTokenResponse, error) {
 	prj, err := s.appclientset.ArgoprojV1alpha1().AppProjects(s.ns).Get(q.Project, metav1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
-	err = prj.ValidateProject()
+	err = validateProject(prj)
 	if err != nil {
 		return nil, err
 	}
@@ -104,7 +116,7 @@ func (s *Server) DeleteToken(ctx context.Context, q *project.ProjectTokenDeleteR
 	if err != nil {
 		return nil, err
 	}
-	err = prj.ValidateProject()
+	err = validateProject(prj)
 	if err != nil {
 		return nil, err
 	}
@@ -141,7 +153,7 @@ func (s *Server) Create(ctx context.Context, q *project.ProjectCreateRequest) (*
 		return nil, err
 	}
 	q.Project.NormalizePolicies()
-	err := q.Project.ValidateProject()
+	err := validateProject(q.Project)
 	if err != nil {
 		return nil, err
 	}
@@ -200,7 +212,7 @@ func (s *Server) Update(ctx context.Context, q *project.ProjectUpdateRequest) (*
 		return nil, err
 	}
 	q.Project.NormalizePolicies()
-	err := q.Project.ValidateProject()
+	err := validateProject(q.Project)
 	if err != nil {
 		return nil, err
 	}

